ext: stm32cube: stm32f4xx: shift I2SR field in PLLI2SCFGR register

avisconti opened the pull request:
This PR is fixing two bugs into ST CubeMX code for stm32f4xx and stm32f7xx. Both the two bugs are related to the routines to configure the PLLI2S where the PLLR field was not properly shifted.

ST BUG Tracker IDs:  50108 and 50086

avisconti committed:
ext: stm32cube: stm32f4xx: shift I2SR field in PLLI2SCFGR register

The I2SR field should be shifted by RCC_PLLI2SCFGR_PLLI2SR_Pos when the
PLLI2SCFGR register is read or written. Without this patch the PLLI2S
configuration is not done properly (R and M params are badly set) and
the PLLI2S generates bad clock waveform.

ST Bug tracker ID: 50086

Signed-off-by: Armando Visconti <armando.visconti@st.com>

avisconti commented:
@erwango 
Please note that for the moment in README file I put:
    ST Bug tracker ID: TO BE ASSIGNED

It would be nice if we could get a number before the end of the week.

erwango commented on ext/hal/st/stm32cube/stm32f4xx/README:
```diff
@@ -43,3 +43,10 @@ Patch List:
+     is read or written.
+    Impacted files:
+     drivers/include/stm32f4xx_ll_rcc.h
+    ST Bug tracker ID: TO BE ASSIGNED
```
50086

avisconti replied:
>> Provided internal ST tracker.

ok, good.

>> Do you know if this bug is also present for other series

Well, no, I don't know.
I can check it quickly tomorrow. Let's say that the PLLI2S is not present in all the series if I got it properly. Maybe only stm32f4xx and stm32f10x. But I need to check carefully.

erwango reviewed (commented):
Provided internal ST tracker.
Do you know if this bug is also present for other series?

avisconti commented:
@erwango 
I think that also stm32f7xx has same bug

erwango commented:
@avisconti , could you create the same patch in this PR for F7 series as well?
Bug tracker : 50108

Thanks!

avisconti committed:
ext: stm32cube: stm32f7xx: shift I2SR field in PLLI2SCFGR register

The I2SR field should be shifted by RCC_PLLI2SCFGR_PLLI2SR_Pos when the
PLLI2SCFGR register is read or written. Without this patch the PLLI2S
configuration is not done properly (R and M params are badly set) and
the PLLI2S generates bad clock waveform.

ST Bug tracker ID: 50108

Signed-off-by: Armando Visconti <armando.visconti@st.com>
